File: vdp-ruby/test/vctc/validation_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class ValidationAPITest < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
  end
  
  def test_retrieveTransactionType
    base_uri = 'vctc/'
    resource_path = 'validation/v1/decisions/history'
    query_param = '?limit=1&page=1'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}#{query_param}", "Retrieve List of Recent Decision Records test", "get", '')
    assert_equal("200", response_code, "Retrieve List of Recent Decision Records test failed")
  end
end

File: vdp-node/test/vctc/validationtest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Visa Consumer Transaction Controls Validation API test', function() {
	var visaAPIClient = new VisaAPIClient();
	
	it('Retrieve List of Recent Decision Records',function(done) {
		this.timeout(10000);
		var baseUri = 'vctc/';
		var resourcePath = 'validation/v1/decisions/history';
		var queryParam = '?limit=1&page=1';
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath + queryParam, '', 'GET', {}, 
		function(err, responseCode) {
		    if(!err) {
		    	assert.equal(responseCode, 200);
		    } else {
		        assert(false);
		    }
		    done();
		});
	});
});
